spec/support/fake_saml_request.rb

```diff
@@ -3,6 +3,10 @@ def service_provider
     self
   end
 
+  def matching_cert
```

Comment:
I hate this but currently we set `FakeSamlRequest#service_provider` to the instance of `FakeSamlRequest` and I didn't want to go down a rabbit hole untangling it (I tried using `instance_double` but ran into RSpec loading issues).